modules/apps/frontend-data-set/frontend-data-set-web/test/utils/resolveField.ts

```diff
@@ -0,0 +1,300 @@
+				name: 'Test Test',
+				userGroupBriefs: [
+					{
+						id: 123,
+						name: 'A team name',
+					},
+				],
+			},
+			dateCreated: '2024-04-16T13:50:57Z',
+			eventType: 'UPDATE',
+		},
+		{
+			auditFieldChanges: [
+				{
+					name: 'description',
+					newValue: 'My custom object is custom',
+					oldValue: 'My custom object',
+				},
+			],
+			creator: {
+				additionalName: '',
+				contentType: 'UserAccount',
+				familyName: 'Test',
+				givenName: 'Test',
+				id: 20122,
+				image:
+					'/image/user_portrait?img_id=38678&img_id_token=7NJsMSX85wO4BAtets1FsmiCKXE%3D&t=1713281378882',
+				name: 'Test Test',
+				userGroupBriefs: [],
+			},
+			dateCreated: '2024-04-16T13:50:41Z',
+			eventType: 'UPDATE',
+		},
+		{
+			auditFieldChanges: [
+				{
+					name: 'image',
+					newValue: {
+						dlFileEntryId: 39346,
+						title: 'trooper',
+					},
+					oldValue: {
+						dlFileEntryId: 39312,
+						title: 'rooper.png',
+					},
+				},
+			],
+			creator: {
+				additionalName: '',
+				contentType: 'UserAccount',
+				familyName: 'Test',
+				givenName: 'Test',
+				id: 20122,
+				image:
+					'/image/user_portrait?img_id=38678&img_id_token=7NJsMSX85wO4BAtets1FsmiCKXE%3D&t=1713281378882',
+				name: 'Test Test',
+				userGroupBriefs: [],
+			},
+			dateCreated: '2024-04-16T13:50:03Z',
+			eventType: 'UPDATE',
+		},
+	],
+	creator: {
+		additionalName: '',
+		contentType: 'UserAccount',
+		familyName: 'Test',
+		givenName: 'Test',
+		id: 20122,
+		image:
+			'/image/user_portrait?img_id=38678&img_id_token=7NJsMSX85wO4BAtets1FsmiCKXE%3D&t=1713281378882',
+		name: 'Test Test',
+	},
+	dateCreated: '2024-04-12T08:28:45Z',
+	dateModified: '2024-04-16T13:50:57Z',
+	description: 'My custom object is custom',
+	description_i18n: {
+		en_US: 'My custom object is custom',
+	},
+	externalReferenceCode: '0b31f896-94e4-8519-ca5b-c74bb374036d',
+	id: 38884,
+	image: {
+		id: 39346,
+		link: {
+			href:
+				'/documents/20119/0/trooper+%281%29.png/a0f947f9-68c8-2410-c6eb-85de1aa70b4f?version=1.0&t=1713171423854&download=true&objectDefinitionExternalReferenceCode=aab85379-959d-b246-0907-e723df904044&objectEntryExternalReferenceCode=0b31f896-94e4-8519-ca5b-c74bb374036d',
+			label: 'trooper (1).png',
+		},
+		name: 'trooper (1).png',
+	},
+	keywords: ['array', 'list', 'collection'],
+	list: [
+		{
+			key: 'one',
+			name: 'One',
+		},
+		{
+			key: 'two',
+			name: 'Two',
+		},
+	],
+	nested: {
+		object: {
+			property: 'hello',
+		},
+	},
+	status: {
+		code: 0,
+		label: 'approved',
+		label_i18n: 'Approved',
+	},
+	taxonomyCategoryBriefs: [],
+	title: 'The first one',
+	title_i18n: {
+		en_US: 'The first one',
+	},
+};
+
+describe('resolveField', () => {
+	it('is defined', () => {
+		expect(resolveField).toBeDefined();
+	});
+
+	describe('single property request', () => {
+		it('returns an object containing the name of the field requested, the item itself and the root property of the object requested', () => {
```

Comment:
I'm not happy with the test descriptions :shrug: . At least, each `given` variable offers a clear example of how each type should behave.